core/runtime/binaryen/module/wasm_module_impl.cpp

```diff
@@ -26,38 +26,47 @@ OUTCOME_CPP_DEFINE_CATEGORY(kagome::runtime::binaryen,
 namespace kagome::runtime::binaryen {
 
   WasmModuleImpl::WasmModuleImpl(
-      std::unique_ptr<wasm::Module> module,
-      std::unique_ptr<wasm::ModuleInstance> module_instance)
+      std::unique_ptr<wasm::Module>&& module,
+      std::unique_ptr<wasm::ModuleInstance>&& module_instance)
       : module_{std::move(module)},
         module_instance_{std::move(module_instance)} {
     BOOST_ASSERT(module_ != nullptr);
     BOOST_ASSERT(module_instance_ != nullptr);
   }
 
+  WasmModuleImpl::~WasmModuleImpl() {
+    // Enforce destruction order because module instance refers to module
+    module_instance_.reset();
+    module_.reset();
```

Comment:
This redundant. Members' destructors being called  in reverse order of declaration.

Reply:
Well, their order of declaration is not set in stone and someone may accidentally change it in future, so I suppose that if we already have this enforcement, it should be okay to keep it?